client/session.go

```diff
@@ -1403,79 +1404,113 @@ func (s *session) selectBlocksForSeriesFromPeerBlocksMetadata(
+						append(currStart[i].blocks[idx].reattempt.attempted, peer)
+					currStart[i].blocks[idx].reattempt.peersMetadata = peersMetadata
+
+					// Leave the block in the current peers blocks list
+					continue
+				}
 
-		// Order by least outstanding blocks being fetched
-		blocksMetadataQueues = blocksMetadataQueues[:0]
-		for i := range currEligible {
-			insert := blocksMetadataQueue{
-				blocksMetadata: currEligible[i],
-				queue:          peerQueues.findQueue(currEligible[i].peer),
+				// Removing this block
+				blocksLen := len(currStart[i].blocks)
+				idx := currStart[i].idx
+				tailIdx := blocksLen - 1
+				currStart[i].blocks[idx], currStart[i].blocks[tailIdx] = currStart[i].blocks[tailIdx], currStart[i].blocks[idx]
+				currStart[i].blocks = currStart[i].blocks[:blocksLen-1]
 			}
-			blocksMetadataQueues = append(blocksMetadataQueues, insert)
-		}
-		sort.Stable(blocksMetadatasQueuesByOutstandingAsc(blocksMetadataQueues))
-
-		// Select the best peer
-		bestPeerBlocksQueue := blocksMetadataQueues[0].queue
-
-		// Prepare the reattempt peers metadata
-		peersMetadata := make([]blockMetadataReattemptPeerMetadata, 0, len(currStart))
-		for i := range currStart {
-			unselected := currStart[i].unselectedBlocks()
-			metadata := blockMetadataReattemptPeerMetadata{
-				peer:     currStart[i].peer,
-				start:    unselected[0].start,
-				size:     unselected[0].size,
-				checksum: unselected[0].checksum,
+		} else {
+			// Prepare the reattempt peers metadata
+			peersMetadata := make([]blockMetadataReattemptPeerMetadata, 0, len(currStart))
+			for i := range currStart {
+				unselected := currStart[i].unselectedBlocks()
+				metadata := blockMetadataReattemptPeerMetadata{
+					peer:     currStart[i].peer,
+					start:    unselected[0].start,
+					size:     unselected[0].size,
+					checksum: unselected[0].checksum,
+				}
+				peersMetadata = append(peersMetadata, metadata)
 			}
-			peersMetadata = append(peersMetadata, metadata)
-		}
 
-		// Remove the block from all other peers and increment index for selected peer
-		for i := range currStart {
-			peer := currStart[i].peer
-			if peer == bestPeerBlocksQueue.peer {
+			for i := range currStart {
 				// Select this block
 				idx := currStart[i].idx
 				currStart[i].idx = idx + 1
 
 				// Set the reattempt metadata
 				currStart[i].blocks[idx].reattempt.id = currID
-				currStart[i].blocks[idx].reattempt.attempt++
-				currStart[i].blocks[idx].reattempt.attempted =
-					append(currStart[i].blocks[idx].reattempt.attempted, peer)
 				currStart[i].blocks[idx].reattempt.peersMetadata = peersMetadata
 
-				// Leave the block in the current peers blocks list
-				continue
+				// Each block now has attempted multiple peers
+				for j := range currStart {
+					currStart[i].blocks[idx].reattempt.attempt++
+					currStart[i].blocks[idx].reattempt.attempted =
+						append(currStart[i].blocks[idx].reattempt.attempted, currStart[j].peer)
```

Comment:
At some point we should rethink our retry semantics here.  If any of these fail they won't be retried against the peer we want them from. 

Let me open an issue about it.  For now this is fine, but we probably want real retry semantics in the future for trying to retrieve from the same peer on failure.